### PR TITLE
Fix issue with setting IP address in run time.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
Issue: https://github.com/Azure/SONiC/issues/22
Issue was observed only when setting IP address via ifconfig
command. Caused by ifconfig command behavior: first is sets
IP address with netmask /8 and then changes netmask to specified
in command. Problem is not observed when set IP address with
"ip addr" command.

### CODE REVIEW

**Comment by stcheng** [comment]
thanks for submitting this fix. i have once checked the ifconfig behavior and found this wired routine but didn't find a food solution of it. have you checked what's the time gap between the nlmsgs? because orchagent has retry logic while trying to apply tasks repeatedly, how many times of the `overlaps` logs have you seen during your test?

File: orchagent/intfsorch.cpp [hunk 1]
```diff
@@ -87,33 +87,53 @@ void IntfsOrch::doTask(Consumer &consumer)
...
-            else
+
+            if (m_syncdIntfses[alias].ip_addresses.count(ip_prefix))
+            {
                 /* Duplicate entry */
                 it = consumer.m_toSync.erase(it);
+                continue;
+            }
+
+            bool overlaps = false;
```
Comment (stcheng) [inline orchagent/intfsorch.cpp]: could you add some comments here to about the `ifconfig` issue? it helps.

Comment (oleksandrivantsiv) [inline orchagent/intfsorch.cpp]: sure

**Comment by oleksandrivantsiv** [comment]
@stcheng time frame between netlink messages is quite small. During tests I saw from one to four messages.

**Comment by stcheng** [comment]
Thanks. Could you also try to resolve the `cla-required` tag?

File: orchagent/intfsorch.cpp [hunk 2]
```diff
@@ -87,33 +87,61 @@ void IntfsOrch::doTask(Consumer &consumer)
...
+            {
                 /* Duplicate entry */
                 it = consumer.m_toSync.erase(it);
+                continue;
+            }
+
+            /* NOTE: Overlap checking is required to handle ifconfig weird behavior.
+             * When set IP address using ifconfig command it applies it in two stages.
+             * On stage one it sets IP address with netmask /8. On stage two it
+             * changes netmask to specified in command. As DB is async event to
```
Comment (qiluo-msft) [inline orchagent/intfsorch.cpp]: >specified in command [](start = 34, length = 20)

Did you test ifconfig with a smaller netmask, such as /7 or /8?

**Pull request merged**